src/Algebra/Structures.agda

```diff
@@ -425,6 +425,15 @@ record IsCommutativeSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ) where
     }
 
 
+record IsCancellativeCommutativeSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ) where
+  field
+    isCommutativeSemiring : IsCommutativeSemiring + * 0# 1#
+    cancelNZˡ             : AlmostLeftCancellative 0# *
```

Comment:
The name of this should change to `*-cancelˡ` or `*-almostCancelˡ`

Reply:
Please, consider this point once more.
I think, ``*-cancelNonzeroˡ`` is better.
This field means the law of ``cancelling by any nonzero element from left``. It is not cancelling by a non-unity, nor cancelling by any positive element, and so on. 
For the signature, AlmostLeftCancellative is all right. But when a reader reads in the code  ``(*-cancelNonzeroˡ ...)`` it is easier to understand than  ``(*-almostCancelˡ ...)``. Because one thinks "what kind of almost?". Then, one has a chance to  guess  of this - by observing the further arguments like ``x/=0`` -  if  such is sensibly denoted. 
?

Reply:
Hmm I understand your point. How about `*-cancelˡ-nonZero` or `*-cancelˡ-≉0`?

Reply:
I set there ``*-cancelˡ-nonZero``
(though I like   ``*-cancelNonzeroˡ``).
Also I run 
```
> cd agda-stdlib-fork
> fix-whitespace. 
```
(before ``make clean; make test ...``). Why does not it work completely, I wonder.
May be it is needed  ``fix-agda-whitespace`` ?
I tried now, and its messages show several fixed places.

Maybe HACKING.ms needs correction at this point?

So I am going to repeat my today's commit.

Another point: I understand this so that ``*-cancelˡ-nonZero`` is the only request of yours, so that after setting 
``*-cancelˡ-nonZero``, the commit is accepted.
Right?
So, I do now the next commit, adding  Algebra.Properties.Magma, Algebra.Properties.Monoid.
They define the divisibility relation and prove a couple of its properties.
They are needed to define irreducibiity and GCD. 
 Further there will be neded more proofs in these two modules and in
 Algebra.Properties.CommutativeMonoid, Algebra.Properties.Semiring,  
Algebra.Properties.CancellativeCommutativeSemiring.